### PR TITLE
remove crate client from test cluster and server

### PR DESCRIPTION
**Pull request by kovrus**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ jdk:
 
 sudo: false
 
-script: ./gradlew test -s
+script: ./gradlew test -s --info
```
Comment (dobe) [inline .travis.yml]: is this by inent?

File: src/main/java/io/crate/testing/CrateTestCluster.java [hunk 2]
```diff
@@ -209,32 +217,24 @@ public static CrateTestCluster cluster(String clusterName,
...
                         }
-                    } catch (NoNodeAvailableException e) {
+                    } catch (IOException e) {
                         // carry on
                     } catch (Exception e) {
                         e.printStackTrace();
                         return false;
                     }
-                    Thread.sleep(100);
+                    Thread.sleep(2000);
```
Comment (dobe) [inline src/main/java/io/crate/testing/CrateTestCluster.java]: try waiting incremental

File: src/main/java/io/crate/testing/CrateTestCluster.java [hunk 3]
```diff
@@ -249,6 +249,38 @@ public Boolean call() throws Exception {
...
+        String query = "{\"stmt\": \"select count(*) as nodes from sys.nodes\"}";
+        byte[] body = query.getBytes("UTF-8");
+        connection.setRequestProperty("Content-Type", "application/text");
+        connection.setRequestProperty("Content-Length", String.valueOf(body.length));
+        connection.setDoOutput(true);
+        connection.getOutputStream().write(body);
+
+        if (connection.getResponseCode() == HttpURLConnection.HTTP_OK) {
+            JsonObject response = parseResponse(connection.getInputStream());
+            return servers.length == response.getAsJsonArray("rows").getAsInt();
```
Comment (dobe) [inline src/main/java/io/crate/testing/CrateTestCluster.java]: i don't really understand this, shouldnt there be a index lookup on the array?

File: src/main/java/io/crate/testing/Utils.java [hunk 4]
```diff
@@ -41,7 +48,59 @@ static int randomAvailablePort() {
         }
     }
 
-    static void log(String message, Object ... params) {
+    static void log(String message, Object... params) {
         System.out.println(String.format(Locale.ENGLISH, message, params));
     }
+
+    public static void deletePath(Path path) throws Exception {
```
Comment (dobe) [inline src/main/java/io/crate/testing/Utils.java]: isnt there something like this in java

Comment (kovrus) [inline src/main/java/io/crate/testing/Utils.java]: we can use common io FileUtils to delete files in the path recursively, but i didn't want to add additional dependencies, just for one method call. With plain java we cannot delete non-empty dirs, thus this file visitor for deleting dir recursively was implemented.

File: .travis.yml (hunk 1 above)
Comment (kovrus) [inline .travis.yml]: gradle output was quite poor, it seemed to be quite helpful for me. But it is fine if you want remove it.

File: src/main/java/io/crate/testing/CrateTestCluster.java (hunk 3 above)
Comment (kovrus) [inline src/main/java/io/crate/testing/CrateTestCluster.java]: the array always has a single element. getAsInt() is applicable in such cases.
```
  public int getAsInt() {
    if (elements.size() == 1) {
      return elements.get(0).getAsInt();
    }
    throw new IllegalStateException();
  }
```

File: src/main/java/io/crate/testing/CrateTestCluster.java [hunk 5]
```diff
@@ -134,7 +139,12 @@ public Builder keepWorkingDir(boolean keepWorkingDir) {
         }
 
         public CrateTestCluster build() {
-            Preconditions.checkArgument(clusterName != null, "no cluster name given");
+            if (clusterName == null) {
```
Comment (dobe) [inline src/main/java/io/crate/testing/CrateTestCluster.java]: we should not require a cluster name

File: src/main/java/io/crate/testing/CrateTestCluster.java [hunk 6]
```diff
@@ -134,7 +139,12 @@ public Builder keepWorkingDir(boolean keepWorkingDir) {
...
+                throw new IllegalArgumentException("no cluster name is given");
+            }
+            if (numberOfNodes <= 0) {
```
Comment (dobe) [inline src/main/java/io/crate/testing/CrateTestCluster.java]: can you check such stuff in the setter, this is more convinient for the user to find the cause

**Comment by dobe** [comment]
ack

**Pull request merged**